apps/frontend/src/store/data_filters.ts

```diff
@@ -353,18 +368,67 @@ export const FilteredDataModule = getModule(FilteredData);
  * - defaulting "omit_overlayed_controls"
  */
 export function filter_cache_key(f: Filter) {
-  // fix the search term
-  let newSearch: string;
-  if (f.searchTerm !== undefined) {
-    newSearch = f.searchTerm.trim();
-  } else {
-    newSearch = '';
-  }
-
   const newFilter: Filter = {
-    searchTerm: newSearch,
+    searchTerm: f.searchTerm?.trim() || '',
     omit_overlayed_controls: f.omit_overlayed_controls || false,
     ...f
   };
   return JSON.stringify(newFilter);
 }
+
+export function filterControlsBy(
+  controls: readonly context.ContextualizedControl[],
+  filters: Record<string, Array<string> | undefined>
+): readonly context.ContextualizedControl[] {
+  if (
+    Object.keys(filters).every(
+      (filter) => !filters[filter] || filters[filter]?.length === 0
+    )
+  ) {
+    return controls;
+  } else {
+    const newFilters: Record<string, Array<string> | undefined> = {};
+    Object.keys(filters).forEach((filter) => {
+      if (filters[filter] && filters[filter]?.length !== 0) {
+        newFilters[filter] = filters[filter];
+      }
+    });
+    return controls.filter((control) => {
+      return Object.keys(newFilters).every((filter) => {
+        const item: string | string[] = _.get(control, filter);
+        if (typeof item === 'string') {
+          return newFilters[filter]?.some((term) =>
+            item.toLowerCase().includes(term.toLowerCase())
+          );
+        } else {
+          return newFilters[filter]?.some((term) =>
+            item.some((value) =>
+              value.toLowerCase().includes(term.toLowerCase())
+            )
+          );
+        }
+      });
```

Comment:
The check here seems inverted, for both code paths you call `return newFilters[filter]?.some((term) =>`, only the inner statement requires an if else.